crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -343,4 +340,45 @@ mod tests {
             "something is wrong with the this test or the code reloading scenes since the relationship between scene entities is broken"
         );
     }
+
+    #[test]
+    fn observers_should_see_mapped_entities() {
+        // Testing that Observers don't see the entites from the scene, but rather from the world itself.
+
+        let mut app = App::new();
+        app.init_resource::<AppTypeRegistry>()
+            .register_type::<Parent>()
+            .observe(|trigger: Trigger<OnAdd, Parent>, query: Query<&Parent>| {
+                let parent = query
+                    .get(trigger.entity())
+                    .expect("entity should have a parent");
+
+                assert_ne!(parent.get(), Entity::PLACEHOLDER);
+            });
+
+        let parent = DynamicEntity {
+            entity: Entity::PLACEHOLDER,
+            components: vec![],
+        };
+
+        let child = DynamicEntity {
+            entity: Entity::try_from(
+                Identifier::new(42, 7, IdKind::Entity).expect("Identifier should be valid"),
+            )
+            .expect("Entity should be valid"),
+            // Bit of a hack to construct a Parent - its FromWorld implementation uses Entity::PLACEHOLDER.
+            components: vec![Box::new(Parent::from_world(app.world_mut()))],
+        };
```

Comment:
This is all slightly hacky, and may be better off being extracted from another world with a few older dummy entities that are left out of the scene.

Reply:
Switched to other-world-extraction to keep it in line with how the corresponding `Scene` test works.